presto-main/src/main/java/com/facebook/presto/operator/scalar/ZipFunction.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.annotation.UsedByGeneratedCode;
+import com.facebook.presto.metadata.BoundVariables;
+import com.facebook.presto.metadata.FunctionRegistry;
+import com.facebook.presto.metadata.Signature;
+import com.facebook.presto.metadata.SqlScalarFunction;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.block.BlockBuilderStatus;
+import com.facebook.presto.spi.block.InterleavedBlockBuilder;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.TypeManager;
+import com.facebook.presto.type.RowType;
+import com.google.common.collect.ImmutableList;
+
+import java.lang.invoke.MethodHandle;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+import static com.facebook.presto.util.ImmutableCollectors.toImmutableList;
+import static com.facebook.presto.util.Reflection.methodHandle;
+import static java.lang.String.join;
+import static java.lang.invoke.MethodType.methodType;
+
+public final class ZipFunction
+        extends SqlScalarFunction
+{
+    private static final int MIN_ARITY = 2;
+    private static final int MAX_ARITY = 4;
+    public static final ZipFunction[] ZIP_FUNCTIONS;
+
+    private static final String FUNCTION_NAME = "zip";
+    private static final String DESCRIPTION = "Aggregate elements of the arrays passed as arguments into rows. The M-th " +
```

Comment:
Got it.
When I saw https://github.com/prestodb/presto/blob/master/presto-docs/src/main/sphinx/functions/array.rst, I thought the documentation would be automatically generated because I saw nothing under "Array Functions" (haven't really checked  though).
After opening the file I noticed that the content is there already but github was just unable to render :P